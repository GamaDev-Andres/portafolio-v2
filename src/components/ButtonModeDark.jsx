/* eslint-disable multiline-ternary */
import { useState } from 'react'

const ButtonModeDark = () => {
  const [isDark, setIsDark] = useState(
    () => localStorage.getItem('isDark') || 'false'
  )
  const toogleModeDark = () => {
    localStorage.setItem('isDark', isDark === 'true' ? 'false' : 'true')
    setIsDark(isDark === 'true' ? 'false' : 'true')
    document.documentElement.classList.toggle('dark')
  }

  return (
    <button
      onClick={toogleModeDark}
      className=" z-20 absolute top-2 bottom-2 right-4 sm:static "
    >
      {isDark === 'false' ? (
        <i className="w-5 h-5 text-black fa-solid fa-moon"></i>
      ) : (
        <i className="text-letra-dark w-5 h-5 fa-solid fa-sun"></i>
      )}
    </button>
  )
}

export default ButtonModeDark
