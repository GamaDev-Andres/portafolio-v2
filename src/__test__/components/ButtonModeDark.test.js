/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import ButtonModeDark from '../../components/ButtonModeDark'

describe('probando <ButtonModeDark/>', () => {
  let wrapper = Enzyme.mount(
    <ButtonModeDark />
  )
  beforeEach(() => {
    wrapper = Enzyme.mount(<ButtonModeDark />)
    localStorage.clear()
  })
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button').exists()).toBe(true)
  })
  test('debe montarse correctamente en modo dark y no dark', () => {
    localStorage.setItem('isDark', 'true')
    wrapper = Enzyme.mount(<ButtonModeDark />)

    expect(wrapper).toMatchSnapshot()

    expect(wrapper.find('i').hasClass('text-letra-dark')).toBe(true)
  })
  test('debe de cambiar el localstorage y poner modo dark cuando hace click', () => {
    localStorage.setItem('isDark', 'true')
    wrapper = Enzyme.mount(<ButtonModeDark />)

    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.find('i').hasClass('text-letra-dark')).toBe(false)
  })
})
