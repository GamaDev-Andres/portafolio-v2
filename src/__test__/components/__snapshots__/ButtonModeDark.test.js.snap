// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <ButtonModeDark/> debe montarse correctamente 1`] = `
<button
  className=" z-20 absolute top-2 bottom-2 right-4 sm:static "
  onClick={[Function]}
>
  <i
    className="w-5 h-5 text-black fa-solid fa-moon"
  />
</button>
`;

exports[`probando <ButtonModeDark/> debe montarse correctamente en modo dark y no dark 1`] = `
<button
  className=" z-20 absolute top-2 bottom-2 right-4 sm:static "
  onClick={[Function]}
>
  <i
    className="text-letra-dark w-5 h-5 fa-solid fa-sun"
  />
</button>
`;
